video/mp_image.c

```diff
@@ -544,11 +544,22 @@ void mp_image_copy_attributes(struct mp_image *dst, struct mp_image *src)
                 memcpy(dst->planes[1], src->planes[1], AVPALETTE_SIZE);
         }
     }
+    assert(src != dst);
     assign_bufref(&dst->icc_profile, src->icc_profile);
     assign_bufref(&dst->dovi, src->dovi);
     assign_bufref(&dst->dovi_buf, src->dovi_buf);
     assign_bufref(&dst->film_grain, src->film_grain);
     assign_bufref(&dst->a53_cc, src->a53_cc);
+
+    assert(!dst->num_ff_side_data || dst->ff_side_data);
+    for (int n = 0; n < dst->num_ff_side_data; n++)
+        av_buffer_unref(&dst->ff_side_data[n].buf);
```

Comment:
I know this is correct but it's somehow unwieldy, why not just:
```cpp
    if (dst->ff_side_data) {
        for (int n = 0; n < dst->num_ff_side_data; n++)
            av_buffer_unref(&dst->ff_side_data[n].buf);
    }
```

Reply:
I just don't see a reason to define behavior for the case that is clearly undefined behavior in most implementation. If the array size is non zero and you have a null pointer, there is something very wrong elsewhere not here.